lib/src/collection/treeset.dart

```diff
@@ -627,10 +625,10 @@ class AvlTreeSet<V> extends TreeSet<V> {
    * See [Set.retainAll]
    */
   void retainAll(Iterable<Object> elements) {
-    List<V> chosen = [];
+    List<V> chosen = <V>[];
     for (var target in elements) {
       if (contains(target)) {
-        chosen.add(target);
+        chosen.add(target as V);
```

Comment:
Yeah, and the cast is then unnecessary.
(It's a shame that the interface methods take a `Iterable<Object>` though).